### PR TITLE
Migrate Service to expose properties

### PR DESCRIPTION
**Pull request by JakeWharton**
These remain binary compatible but do not include source-compatible trampolines.

Has been annoying me for weeks. Finally got around to it.

### CODE REVIEW

File: wire-library/wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/Service.kt [hunk 1]
```diff
@@ -21,46 +21,28 @@ import com.squareup.wire.schema.ProtoMember.Companion.get
...
 import kotlin.jvm.JvmStatic
 
 data class Service(
-  private val protoType: ProtoType,
-  private val location: Location,
-  private val documentation: String,
-  private val name: String,
-  private val rpcs: List<Rpc>,
-  private val options: Options
+  @get:JvmName("type") // For binary compatibility.
```
Comment (swankjesse) [inline wire-library/wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/Service.kt:28]: honest question: do we care?

Comment (swankjesse) [reply wire-library/wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/Service.kt:28]: (do we care about binary compatibility for wire-schema?)

Comment (JakeWharton) [reply wire-library/wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/Service.kt:28]: I don't know. I maintained it in the past for a similar migration. It's low effort here so might as well leave it, but I don't really know moving forward. Doesn't Jackson depend on it?

**Comment by oldergod** [comment]
Thank you!

**Pull request merged**